### ISSUE TITLE
Unclear exception and wrong exceptions order

### ISSUE THREAD

**Issue by PhantomYdn** [orientechnologies/orientdb#7538]
### Java Version: 8+ 

I have the following SQL:

```sql
insert into Report from select sum(amount) as value, project, expedentureType as account from Payment group by project, expedentureType
```
Execution failed, but instead of meaningful exception I see the following:

```java
$ANSI{green {db=Orienteer}} Exception `5067255D` in storage `Orienteer`
com.orientechnologies.orient.core.exception.OConfigurationException: Cannot undo the document because tracking of changes is disabled
	DB name="Orienteer"
	at com.orientechnologies.orient.core.record.impl.ODocument.undo(ODocument.java:1588)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2740)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:102)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1805)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1796)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:63)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.saveRecord(OCommandExecutorSQLInsert.java:330)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLInsert.result(OCommandExecutorSQLInsert.java:289)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.pushResult(OCommandExecutorSQLResultsetAbstract.java:285)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.getResult(OCommandExecutorSQLResultsetAbstract.java:265)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execute(OCommandExecutorSQLSelect.java:495)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:74)
```

By digging into the code I found a place where real exception is being lost:

```java
//ODocument line 2737
    try {
      doc.validate();
    } catch (OValidationException ex) {
      doc.undo();
      throw ex;
    }
```

As you can see error during `doc.undo()` overriding real exception.
In my case validation error was about wrong type of a link for field `account`.
Please consider making original exception available for a user instead of secondary generated.

**Comment by tglman** [comment]
hi @PhantomYdn,

Do you have an example query that produce this issue ? I understood the problem and I'm aware of how to fix it, but I haven't reproduced it yet, I would prefer to have the test case with for the fix.

Regards.

**Comment by tglman** [comment]
hi @PhantomYdn,

I wrote the test, reproduced and fixed, it will be released with 2.2.24 and 3.0.0, closing this.

Regards

**Comment by marktshaw** [comment]
@luigidellaquila  Has this been released? I'm not finding it any of the release notes. assuming that I didn't miss it.

Thanks!

**Comment by luigidellaquila** [comment]
Hi @marktshaw 

It was released with 2.2.24, probably it was not properly added to the release notes

Thanks

Luigi
